ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -1230,12 +1232,16 @@ else if(statementId != parsedDelta.statementId) {
    * that for any dir, either all files are acid or all are not.
    */
   public static ParsedDelta parsedDelta(Path deltaDir, FileSystem fs) throws IOException {
-    return parsedDelta(deltaDir, fs, null);
+    return parsedDelta(deltaDir, fs, null, false, -1);
   }
 
-  private static ParsedDelta parsedDelta(Path deltaDir, FileSystem fs, HdfsDirSnapshot dirSnapshot)
+  private static ParsedDelta parsedDelta(Path deltaDir, FileSystem fs, HdfsDirSnapshot dirSnapshot,
+      boolean canTrim, long highWaterMark)
       throws IOException {
     ParsedDeltaLight deltaLight = ParsedDeltaLight.parse(deltaDir);
+    if(canTrim && !(deltaLight.minWriteId >= highWaterMark)){
```

Comment:
please comment this part code, as this is the point of this patch as I understood

so according to some circumstances (canTrim + watermark thing, about which I have no idea :) ), getAcidState -> getChildState -> parsedDelta can short-circuit, so I'm assuming this will prevent processing some deltas, leading to performance improvement

Reply:
Yeah, it will not read the deltas that are not part of the current transaction or were written before the current transaction.